torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -286,8 +286,8 @@ auto handle_torch_function_no_python_arg_parser(
       // See https://github.com/pytorch/pytorch/issues/63767
       if (PyObject_FastGetAttrString(torch_function.ptr(), "__self__").is(arg) &&
           torch_function.ptr() != torch::disabled_torch_function_impl()) {
-        TORCH_WARN("Defining your `__torch_function__` as a plain method is deprecated and ",
-                   "will be an error in future, please define it as a classmethod.");
+        TORCH_WARN_ONCE("Defining your `__torch_function__` as a plain method is deprecated and ",
```

Comment:
The corresponding test should be updated to use `assertWarnsOnceRegex`
https://github.com/pytorch/pytorch/blob/92a9c0e3e050ac9e1972914d6040c9dd0c69cbdb/test/test_overrides.py#L1106-L1108